lametro/views.py

```diff
@@ -548,65 +427,29 @@ def get_context_data(self, **kwargs):
-                FILTER (WHERE mm.label is not Null) as label,
-                split_part(p.name, ' ', 2) AS last_name
-              FROM councilmatic_core_membership AS m
-              LEFT JOIN (
-                SELECT
-                  person_id,
-                  array_agg(DISTINCT pt.label) as label
-                FROM councilmatic_core_membership AS m
-                JOIN councilmatic_core_post AS pt
-                  ON m.post_id=pt.ocd_id
-                WHERE m.organization_id = %s
-                GROUP BY person_id
-              ) AS mm
-                USING(person_id)
-              JOIN councilmatic_core_person AS p
-                ON m.person_id = p.ocd_id
-              WHERE m.organization_id = %s
-              AND m.end_date::date > NOW()::date
-              AND m.role != 'Chief Executive Officer'
-              GROUP BY
-                m.role, p.name, p.slug, p.ocd_id, m.extras
-              {order_by}
-            '''
-            committee_sql = base_sql.format(order_by="ORDER BY CASE " +
-                                           "WHEN m.role='Chair' THEN 0 " +
-                                           "WHEN m.role='Vice Chair' THEN 1 " +
-                                           "WHEN m.role='Member' THEN 2" +
-                                           "END")
-
-            cursor.execute(committee_sql, [settings.OCD_CITY_COUNCIL_ID, committee.ocd_id])
-            columns = [c[0] for c in cursor.description]
-            results_tuple = namedtuple('Member', columns)
-            objects_list = [results_tuple(*r) for r in cursor]
-
-            context['membership_objects'] = objects_list
-
-            ad_hoc_committee_sql = base_sql.format(order_by="ORDER BY CASE " +
-                                           "WHEN m.role='Chair' THEN 0 " +
-                                           "WHEN m.role='1st Vice Chair' THEN 1 " +
-                                           "WHEN m.role='2nd Vice Chair' THEN 2 "
-                                           "WHEN m.role='Member' THEN 3" +
-                                           "END")
-
-            cursor.execute(ad_hoc_committee_sql, [settings.OCD_CITY_COUNCIL_ID, committee.ocd_id])
-
-            columns = [c[0] for c in cursor.description]
-            results_tuple = namedtuple('Member', columns)
-            objects_list = [results_tuple(*r) for r in cursor]
-
-            context['ad_hoc_list'] = objects_list
-
-            context['ceo'] = Person.objects.filter(memberships__role='Chief Executive Officer',                                   memberships___organization=committee.ocd_id).first()
+        try:
+            ceo = Membership.objects\
+                .get(post__role='Chief Executive Officer',
+                     end_date_dt__gt=Now())\
+                .person
+        except Membership.DoesNotExist:
+            ceo = None
+
+        non_ceos = committee.all_members\
+            .annotate(index=Case(
```

Comment:
Likewise, this annotation could be a manager method, e.g., `with_display_index` (since you are sorting differently here vs. in line 501). Or, if you don't mind sorting by index and name in both instances, you could roll it all up into `in_display_order`.

Reply:
I'd like to handle this, and some other things in https://github.com/datamade/django-councilmatic/issues/261

Can this wait till then.

Reply:
It seems like this particular annotation is specific to Metro, no?

Reply:
you are right. I'd actually like to handle this in the scraper (through the extras field)

Reply:
Generally, I'm in favor of moving things upstream, but this feels much more related to display than the legislative data itself. Beyond that, it would be nice to keep display order ephemeral, so we can change it without writing and running a database migration. Are you strongly committed to handling it in the scraper?

Reply:
okay, sure. can I handle moving this to the manager in separate PR.

Reply:
sure, makes sense. Can I handle moving the annotations to the manager in a separate PR?

Reply:
Sure thing. Thank you!